test_router_calls_slack_for_configured_workspace — mock never called

## Problem

`tests/test_delivery_router.py::test_router_calls_slack_for_configured_workspace` fails with:

\`\`\`
AssertionError: Expected 'send' to have been called once. Called 0 times.
\`\`\`

## Analysis

The test mocks \`SlackSender.send\`, then invokes the delivery router for a workspace configured with a Slack channel, then asserts \`send\` was called once. The mock is never called — either the router doesn't reach the Slack branch, or the mock is patched at the wrong import path (common Python mock pitfall: patch where it's used, not where it's defined).

## Possible root causes

1. The router's channel-dispatch logic has a bug — maybe it requires additional config beyond just \`delivery_channels["slack"]\`.
2. The mock is patched at \`app.delivery.slack.SlackSender\` when the router imports it as \`from app.delivery.slack import SlackSender\` and uses a reference that bypasses the patched module.
3. The alert model used in the test doesn't match the router's expectations (e.g., priority filter, status filter).

## Acceptance criteria

- [ ] Test passes
- [ ] If the bug is in test (mock patching), fix the test
- [ ] If the bug is in the router, fix the router AND make sure an additional test covers the fixed behavior

## Labels

\`bug\`, \`tech-debt\`, \`pre-existing\`

--- a/backend/tests/test_delivery_router.py
+++ b/backend/tests/test_delivery_router.py
@@ -1,11 +1,11 @@
 import pytest
-from unittest.mock import AsyncMock, patch, MagicMock
+from types import SimpleNamespace
+from unittest.mock import AsyncMock, MagicMock, patch
 from uuid import uuid4
 from app.delivery.router import DeliveryRouter
 
 
 def make_alert(priority: str = "high"):
-    from types import SimpleNamespace
     return SimpleNamespace(
         id=uuid4(),
         workspace_id=uuid4(),
@@ -23,12 +23,24 @@ def make_alert(priority: str = "high"):
     )
 
 
-def make_workspace_with_slack():
-    from types import SimpleNamespace
+def make_workspace_with_slack(min_priority: str = "low"):
     return SimpleNamespace(
         id=uuid4(),
-        delivery_channels={"slack": {"webhook_url": "https://hooks.slack.com/test", "min_priority": "low"}},
+        delivery_channels={
+            "slack": {
+                "webhook_url": "https://hooks.slack.com/test",
+                "min_priority": min_priority,
+            }
+        },
     )
+
+
+def _mock_sender_class():
+    """Build a sender-class mock that returns a mock instance whose .send is an AsyncMock."""
+    mock_instance = MagicMock()
+    mock_instance.send = AsyncMock()
+    mock_class = MagicMock(return_value=mock_instance)
+    return mock_class, mock_instance
 
 
 @pytest.mark.asyncio
@@ -36,33 +48,32 @@ async def test_router_calls_slack_for_configured_workspace():
     alert = make_alert(priority="high")
     workspace = make_workspace_with_slack()
 
-    with patch("app.delivery.router.SlackSender") as MockSlack:
-        mock_instance = MagicMock()
-        mock_instance.send = AsyncMock()
-        MockSlack.return_value = mock_instance
+    mock_class, mock_instance = _mock_sender_class()
 
+    # Patch the CHANNEL_SENDERS dict directly. Patching `app.delivery.router.SlackSender`
+    # does NOT work because CHANNEL_SENDERS was populated at import time with a direct
+    # reference to the original SlackSender class — the dict entry doesn't follow
+    # subsequent rebinding of the module-level name.
+    with patch.dict("app.delivery.router.CHANNEL_SENDERS", {"slack": mock_class}):
         router = DeliveryRouter()
         await router.deliver(alert=alert, workspace=workspace)
 
-        mock_instance.send.assert_called_once()
+    mock_instance.send.assert_called_once()
+    # Verify the sender was called with both alert and workspace
+    _, kwargs = mock_instance.send.call_args
+    assert kwargs["alert"] is alert
+    assert kwargs["workspace"] is workspace
 
 
 @pytest.mark.asyncio
 async def test_router_skips_channel_below_min_priority():
     alert = make_alert(priority="low")
-    workspace_with_high_threshold = __import__("types").SimpleNamespace(
-        id=uuid4(),
-        delivery_channels={
-            "slack": {"webhook_url": "https://hooks.slack.com/test", "min_priority": "high"}
-        },
-    )
+    workspace = make_workspace_with_slack(min_priority="high")
 
-    with patch("app.delivery.router.SlackSender") as MockSlack:
-        mock_instance = MagicMock()
-        mock_instance.send = AsyncMock()
-        MockSlack.return_value = mock_instance
+    mock_class, mock_instance = _mock_sender_class()
 
+    with patch.dict("app.delivery.router.CHANNEL_SENDERS", {"slack": mock_class}):
         router = DeliveryRouter()
-        await router.deliver(alert=alert, workspace=workspace_with_high_threshold)
+        await router.deliver(alert=alert, workspace=workspace)
 
-        mock_instance.send.assert_not_called()
+    mock_instance.send.assert_not_called()

